### PR TITLE
Action on subgroups of `TorQuadModule`: step 1

### PR DESCRIPTION
**Pull request by StevellM**
I have written on a personal repository (related to lattices with isometry) some functions about actions on subgroups of torsion quadratic modules by automorphisms (not conjugation). 

This is an import of the first functions, namely detecting invariant subgroups under the action of an abelian group endomorphism, and restriction of such endomorphisms.

### CODE REVIEW

File: src/Groups/abelian_aut.jl [hunk 1]
```diff
@@ -268,3 +268,137 @@ function embedding_orthogonal_group(i::TorQuadModuleMor)
...
+#  Action on injections
+#
+###############################################################################
+
+@doc raw"""
+    is_invariant(f::TorQuadModuleMor, i::TorQuadModuleMor) -> Bool
+
+Given an abelian group embedding $i\colon S \to T$ of a torsion quadratic module
+`S` in a torsion quadratic module `T`, and an abelian group endomorphism `f`
+of `T`, return whether `f` fixes `S`, i.e. whether $f(i(s)) \in i(S)$ for all
```
Comment (simonbrandhorst) [inline src/Groups/abelian_aut.jl]:
```suggestion
of `T`, return whether `f` preserves `S`, i.e. whether $f(i(s)) \in i(S)$ for all
```

File: src/Groups/abelian_aut.jl [hunk 2]
```diff
@@ -268,3 +268,137 @@ function embedding_orthogonal_group(i::TorQuadModuleMor)
...
+$s \in S$.
+"""
+function is_invariant(f::TorQuadModuleMor, i::TorQuadModuleMor; check::Bool = true)
+  @req !check || is_invariant(i) "i must be an injection"
```
Comment (simonbrandhorst) [inline src/Groups/abelian_aut.jl]: Mathematically, it makes sense also for `i` not injective.

File: src/Groups/abelian_aut.jl [hunk 3]
```diff
@@ -268,3 +268,137 @@ function embedding_orthogonal_group(i::TorQuadModuleMor)
...
+  return true
+end
+
+@doc raw"""
+    is_invariant(f::AutomorphismGroupElem{TorQuadModule}, i::TorQuadModuleMor)
+                                                                        -> Bool
+
+Given an abelian group embedding $i\colon S \to T$ of a torsion quadratic module
+`S` in a torsion quadratic module `T`, and an automorphism `f` of `T`, return
+whether `f` fixes `S`, i.e. whether $f(i(s)) \in i(S)$ for all $s \in S$.
```
Comment (simonbrandhorst) [inline src/Groups/abelian_aut.jl]:
```suggestion
whether `f` preserves `S`, i.e. whether $f(i(s)) \in i(S)$ for all $s \in S$.
```

File: src/Groups/abelian_aut.jl [hunk 4]
```diff
@@ -268,3 +268,137 @@ function embedding_orthogonal_group(i::TorQuadModuleMor)
...
+  return is_invariant(hom(f), i, check = false)
+end
+
+@doc raw"""
+    is_invariant(G::AutomorphismGroup{TorQuadModule}, i::TorQuadModuleMor)
+                                                                        -> Bool
+
+Given an abelian group embedding $i\colon S \to T$ of a torsion quadratic module
+`S` in a torsion quadratic module `T`, and a group `G` of automorphisms of `T`,
+return whether `S` is fixed by every elements in `G`, i.e. whether
```
Comment (simonbrandhorst) [inline src/Groups/abelian_aut.jl]:
```suggestion
return whether `S` is preserved by every element in `G`, i.e. whether
```

**Review by simonbrandhorst — COMMENTED** [review]
Looks good to me. @ThomasBreuer  Does this fit into our philosophy of working with groups?

File: src/Groups/abelian_aut.jl (hunk 2 above)
Comment (ThomasBreuer) [inline src/Groups/abelian_aut.jl]: Does `is_invariant(i)` check injectivity?

Comment (StevellM) [reply src/Groups/abelian_aut.jl]: No indeed, it is a typo, it should have been `is_injective(i)` here.. Thanks!

**Comment by ThomasBreuer** [comment]
@simonbrandhorst 
> Looks good to me. @ThomasBreuer Does this fit into our philosophy of working with groups?

Yes, nice.
(In fact, this "Hecke style" syntax of `is_invariant` and similar functions, using an embedding as the second argument, should become supported also for `GAPGroup`s.)

File: src/Groups/abelian_aut.jl (hunk 2 above)
Comment (StevellM) [reply src/Groups/abelian_aut.jl]: Yes  indeed, if we talk about invariance of the image. The idea for `is_invariant` was really related to invariant subgroups. I could remove the restriction though, but will it be of any use ?

Comment (simonbrandhorst) [reply src/Groups/abelian_aut.jl]: Can't hurt.

**Review by simonbrandhorst — APPROVED** [review]

**Pull request merged**